src/storage/src/table/cell_based_table/iter_utils.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::binary_heap::PeekMut;
+use std::collections::BinaryHeap;
+
+use futures::StreamExt;
+use futures_async_stream::try_stream;
+use risingwave_common::array::Row;
+
+use super::PkAndRowStream;
+use crate::table::cell_based_table::StorageError;
+
+struct Node<S: PkAndRowStream> {
```

Comment:
Shall we add some brief comments here? Something like telling what `Node` is used for is okay.